internal/local/structure_v2.go

```diff
@@ -292,33 +295,62 @@ func parseDataSources(rootDir string) ([]DataSourceStructure, error) {
 
 			colls := directoryWalker{path: dbPath, onlyDirs: true}
 			if err := colls.walk(func(coll os.FileInfo, collPath string) error {
+				// A valid data sources folder contains at least one of:
+				// - a rules.json file
+				// - a pair of files, schema.json and relationships.json
+				// If neither of these conditions are true (e.g. there is only a schema.json
+				// file, or there are no files in the directory at all), we should error
 
-				rulePath := filepath.Join(collPath, FileRules.String())
-				if _, err := os.Stat(rulePath); err != nil {
```

Comment:
[opt] could we use `Stat()` to check the file paths to see if the conditions for a valid data source folder are met before parsing them? up to you, I'm not sure how much of a performance improvement this would be